src/simulation/replanning/replanner.rs

```diff
@@ -0,0 +1,494 @@
+use std::rc::Rc;
+use tracing::debug;
+
+use crate::simulation::id::{Id, IdStore};
+use crate::simulation::messaging::communication::communicators::SimCommunicator;
+use crate::simulation::messaging::events::EventsPublisher;
+use crate::simulation::messaging::messages::proto::{Activity, Agent, Leg};
+use crate::simulation::network::global_network::Network;
+use crate::simulation::network::sim_network::SimNetworkPartition;
+use crate::simulation::population::population::ActType;
+use crate::simulation::replanning::routing::router::Router;
+use crate::simulation::replanning::routing::travel_times_collecting_alt_router::TravelTimesCollectingAltRouter;
+use crate::simulation::replanning::walk_finder::{EuclideanWalkFinder, WalkFinder};
+use crate::simulation::vehicles::garage::Garage;
+
+pub trait Replanner {
+    fn next_time_step(&mut self, now: u32, events: &mut EventsPublisher);
```

Comment:
I still find the naming not very fitting.

Can you use something like 
`next_time_step` -> `handle_next_time_step` or `update_time` or something like this
`update_agent` -> `replan` or `plan` or something which indicates what will happen to the agent.